### PR TITLE
doc: update README and badge images to Node-API

### PR DESCRIPTION
**Pull request by jschlight**
This PR updates the project's README file from N-API to Node-API and removes text related to Node-API being in experimental state. It also updates the content of the badge images from N-API to Node-API without changing the names of the SVG image files.

### CODE REVIEW

**Review by mhdawson — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -52,35 +44,31 @@ It is distributed as a separate npm package: [https://www.npmjs.com/package/node
...
 
-![N-API v1 Badge](assets/N-API%20v1%20Badge.svg)
-![N-API v2 Badge](assets/N-API%20v2%20Badge.svg)
-![N-API v3 Badge](assets/N-API%20v3%20Badge.svg)
-![N-API v4 Badge](assets/N-API%20v4%20Badge.svg)
-![N-API v5 Badge](assets/N-API%20v5%20Badge.svg)
-![N-API v6 Badge](assets/N-API%20v6%20Badge.svg)
-![N-API v7 Badge](assets/N-API%20v7%20Badge.svg)
-![N-API Experimental Version Badge](assets/N-API%20Experimental%20Version%20Badge.svg)
+![Node-API v1 Badge](assets/N-API%20v1%20Badge.svg)
```
Comment (legendecas) [inline README.md]: I was assuming we can put `assets/Node-API%20v1%20Badge.svg` along side with existing `assets/N-API%20v1%20Badge.svg` (those are changed with "Node-API" label in their content too)?

**Review by legendecas — APPROVED** [review]

**Review by NickNaso — COMMENTED** [review]
Hi Jim, 
now we have a series of files on `assets` folder and you changed their content and this is good for me. What do you think to create a new set of files named: `Node-API vX Badges.svg` and for the new badges we will use this new naming convention? We should to mantinain the old files because other devs could use them in their repo.

File: assets/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# About these files
+
+In February 2021, the N-API project was renamed to Node-API.
+
+In order to maintain links that module maintainers may have created to
+these files, the badge files using the old file naming convention have been
+retained. Moving forward, the “Node-API“ naming convention should be used
+for new badge image files.
+
+[renaming n-api to something more appropriate and
```
Comment (NickNaso) [inline assets/README.md]: Could you change the r in uppercase?  Renaming n-api ...

**Review by NickNaso — APPROVED** [review]

**Pull request merged**